plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/SourceRole.kt

```diff
@@ -11,7 +11,7 @@ import org.jetbrains.kotlin.fir.types.ConeKotlinType
 import org.jetbrains.kotlin.formver.viper.ast.Info
 
 sealed interface SourceRole {
-    data object ParamFunctionLeakageCheck : SourceRole
+    data class ParamFunctionLeakageCheck(val functionRole: FirSymbolHolder) : SourceRole
```

Comment:
Well, kind of. Besides, it was really unpleasant to debug this since it wasn't clear enough how the symbol source role is accessed. I hope such a way would be clearer for next generations.

Reply:
That's fair, being explicit here is probably a better idea anyway :+1: